opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -12,6 +12,131 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+"""
+The OpenTelemetry context module provides abstraction layer on top of
+thread-local storage and contextvars. The long term direction is to switch to
+contextvars provided by the Python runtime library.
+
+A global object ``Context`` is provided to access all the context related
+functionalities:
+
+    >>> from opentelemetry.context import Context
+    >>> Context.foo = 1
+    >>> Context.foo = 2
+    >>> Context.foo
+    2
+
+When explicit thread is used, a helper function `Context.with_current_context`
+can be used to carry the context across threads:
+
+    from threading import Thread
+    from opentelemetry.context import Context
+
+    def work(name):
+        print('Entering worker:', Context)
+        Context.operation_id = name
+        print('Exiting worker:', Context)
+
+    if __name__ == '__main__':
+        print('Main thread:', Context)
+        Context.operation_id = 'main'
+
+        print('Main thread:', Context)
+
+        # by default context is not propagated to worker thread
+        thread = Thread(target=work, args=('foo',))
+        thread.start()
+        thread.join()
+
+        print('Main thread:', Context)
+
+        # user can propagate context explicitly
+        thread = Thread(
+            target=Context.with_current_context(work),
```

Comment:
this helps clear up the value of with_current_context. Nice! 

I feel like the need to add with_current_context will probably be a gotcha in many cases. It's too bad there isn't a way to make this something that is implicitly shared.

Reply:
When the day comes that contextvars or some built-in Python lib support such scenario, we can say goodbye to this `Context` class and never need to reinvent wheels again :)